zipline/pipeline/filters/filter.py

```diff
@@ -580,3 +588,52 @@ def compute(self, today, assets, out, value):
                 is_missing(value, self.inputs[0].missing_value),
                 axis=0,
             )
+
+
+class MaximumFilter(Filter, StandardOutputs):
+    """Pipeline filter that selects the top asset, possibly grouped and masked.
+    """
+    window_length = 0
+
+    def __new__(cls, factor, groupby, mask):
+        return super(MaximumFilter, cls).__new__(
+            cls,
+            inputs=(factor, groupby),
+            mask=mask,
+        )
+
+    def _compute(self, arrays, dates, assets, mask):
+        data = arrays[0]
+        groupby_expr = self.inputs[1]
+        if groupby_expr.dtype == int64_dtype:
+            group_labels = arrays[1]
+            null_label = self.inputs[1].missing_value
+        elif groupby_expr.dtype == categorical_dtype:
+            # Coerce our LabelArray into an isomorphic array of ints.  This is
+            # necessary because np.where doesn't know about LabelArrays or the
+            # void dtype.
+            group_labels = arrays[1].as_int_array()
+            null_label = arrays[1].missing_value_code
+        else:
+            raise TypeError(
+                "Unexpected groupby dtype: %s." % groupby_expr.dtype
+            )
+
+        effective_mask = (
+            mask
+            & (group_labels != null_label)
+            & ~is_missing(data, self.inputs[0].missing_value)
+        ).view(uint8)
+
+        if is_datetime(data):
+            data = data.view(float64_dtype)
+
+        return grouped_masked_is_maximal(
+            data,
+            # TODO: Support different sizes of group labels.
```

Comment:
We would if the `astype` started coming up as a real cost, but I don't think that'll happen any time soon.  I'll change this to a `PERF` note saying we might want to consider this instead of a full-blown TODO.